Modules/RatMama.py

```diff
@@ -0,0 +1,176 @@
+    (?P<language_code>.+?)               # "en"
+    (?:                                  # Optional group
+        -(?P<language_country>.+)        # "-", "US" (currently unused)
+    )?                                   # Actually make the group optional.
+    \)                                   # The ")" of "(en-US)"
+    )                                    # End of full language text
+    (?:                                  # Possibly match IRC nickname
+    \s+-\s+                              #  -
+    IRC\s+Nickname:\s*(?P<nick>[^\s]+)   # IRC nickname
+    )?                                   # ... emphasis on "Possibly"
+    )                                    # End of the main capture group
+    \s*                                  # Handle any possible trailing whitespace
+    $                                    # End of pattern
+""")
+
+
+@rule(r"^\s*Incoming Client:", case_sensitive=False, full_message=True, prefixless=True,
+      pass_match=False)
+async def handle_ratmama_announcement(ctx: Context):
+    if ctx.user.nickname.casefold() not in ('ratmama[bot]',):
+        return
+
+    message: str = ctx.words_eol[0]
+    result = re.fullmatch(ratmama_regex, message)
+    client_name: str = result.group("cmdr")
+    system_name: str = result.group("system")
+    platform_name: str = result.group("platform")
+    o2_status: bool = result.group("o2") == "OK"  # false is CR
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! Case #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
+        if system_name.casefold() != exist_rescue.system.casefold():
+            diff_response += f"System changed! "
+
+        if platform_name.casefold() != exist_rescue.platform.name.casefold():
+            diff_response += "Platform changed! "
+
+        if not o2_status != exist_rescue.code_red:
+            diff_response += "O2 Status changed!" if o2_status else\
+                "O2 Status changed, it is now CODE RED!"
+
+        if not diff_response == "":
+            await ctx.reply(diff_response)
+
+    else:
+        platform = None
+
+        if platform_name.casefold() in ("pc", "ps", "xb"):
+            platform = Platforms[platform_name.upper()]
+        else:
+            LOG.warning(f"Got unknown platform from RatMama: {platform_name}")
+
+        # no case for that name, we have to make our own
+        rescue: Rescue = Rescue(client=client_name, system=system_name, irc_nickname=nickname,
+                                code_red=not o2_status, lang_id=lang_code, platform=platform)
+
+        board.append(rescue, overwrite=False)
+        index = board.find_by_name(client=client_name).board_index
```

Comment:
`board_index` is an attribute of the rescue, no? Assuming the rescue is added to the board by reference, shouldn't we be able to retrieve the index simply with `rescue.board_index` instead of doing a text search?

Reply:
yes, we should and we are now.

Reply:
I don't think we are. We're still searching the board again after adding the rescue.